nethack/src/netgame.c

```diff
@@ -713,7 +713,8 @@ netgame(void)
     if (ui_flags.connection_only) {
         servlist = NULL;
         server = &localserver;
-        localserver.hostname = strdup("::1");
```

Comment:
Looks like in https://github.com/liam-middlebrook/bingehack4/blob/f0b00f64b1e7e396c3f34aa5dbcca519b2927cb1/libnethack_client/src/connection.c#L516
it uses a default port if set to 0. I'll change this to use that then.